legistar/bills.py

```diff
@@ -267,12 +267,23 @@ def history(self, matter_id) :
     def sponsors(self, matter_id) :
         spons = self.endpoint('/matters/{0}/sponsors', matter_id)
         if spons:
-            max_version = str(max(int(sponsor['MatterSponsorMatterVersion'])
-                              for sponsor in spons))
-            spons = [sponsor for sponsor in spons
-                     if sponsor['MatterSponsorMatterVersion'] == max_version]
-            return sorted(spons, 
+            try:
+                max_version = str(max(int(sponsor['MatterSponsorMatterVersion'])
+                                  for sponsor in spons))
+
+            except ValueError:
+                if sponsor['MatterSponsorMatterVersion'] == '*':
```

Comment:
Does it ever happen that if there is a version with `*` there is another also another version?

Reply:
How do you mean?

Reply:
This code seems to assume that there is only one version of the matter and that version is '*'

Is that right?

Reply:
Yes. For example, here is the cached version data for the matter referenced in the PR description:

```
status: 200
encoding: utf-8
Cache-Control: no-cache
Pragma: no-cache
Content-Type: application/json; charset=utf-8
Expires: -1
Server: Microsoft-IIS/10.0
X-AspNet-Version: 4.0.30319
X-Powered-By: ASP.NET
GranicusServer: grasp-legapi1
Date: Wed, 01 Nov 2017 20:22:35 GMT
Content-Length: 29

[{"Key":"25377","Value":"*"}]
```

Reply:
Re: your original question, I `grep`ed the cache and found some matters with `*` versions with a second version, either `A` or `0`. So it's not always the only one, it seems.

Reply:
hrmm.. maybe we should do something like this:

```python
...
max_version = str(max(self._version_rank(sponsor['MatterSponsorMatterVersion'])
                                  for sponsor in spons))
...

def _version_rank(self, version):
   return int(verson)
```

and then in the NYC Bill Scraper we would override `_version_rank`?

Reply:
seems reasonable to me